src/geom/face_quad.C

```diff
@@ -200,14 +201,85 @@ Real Quad::quality (const ElemQuality q) const
           return std::sqrt(2) * min_edge / d_max;
       }
 
+    case SHAPE:
+    case SKEW:
+      {
+        // From: P. Knupp, "Algebraic mesh quality metrics for
+        // unstructured initial meshes," Finite Elements in Analysis
+        // and Design 39, 2003, p. 217-241, Sections 5.2 and 5.3.
+        typedef std::array<Real, 4> Array4;
+        typedef std::array<Real, 6> Array6;
+
+        // x, y, z node coordinates.
+        std::vector<Real>
+          x = {this->point(0)(0), this->point(1)(0), this->point(2)(0), this->point(3)(0)},
+          y = {this->point(0)(1), this->point(1)(1), this->point(2)(1), this->point(3)(1)},
+          z = {this->point(0)(2), this->point(1)(2), this->point(2)(2), this->point(3)(2)};
+
+        // Nodal Jacobians. These are 3x2 matrices, hence we represent
+        // them by Array6.
+        std::vector<Array6> A(4);
+        for (unsigned int k=0; k<4; ++k)
+          {
+            unsigned int
+              kp1 = k+1 > 3 ? k+1-4 : k+1,
+              kp3 = k+3 > 3 ? k+3-4 : k+3;
+
+            // To initialize std::array we need double curly braces in
+            // C++11 but not C++14 apparently.
+            A[k] = {{x[kp1] - x[k], x[kp3] - x[k],
+                     y[kp1] - y[k], y[kp3] - y[k],
+                     z[kp1] - z[k], z[kp3] - z[k]}};
+          }
+
+        // Compute metric tensors, T_k = A_k^T * A_k. These are 2x2
+        // square matrices, hence we represent them by Array4.
+        std::vector<Array4> T(4);
```

Comment:
Why std::vector rather than std::array for the outer container here?  We've got a fixed compile-time size.

Reply:
I'm using `std::vector` in some places because of the annoying [double curly brace initialization syntax](http://en.cppreference.com/w/cpp/container/array) that is still required in C++11 (and which clang warns about if you forget), but you're right, this one might as well be a `std::array` as well.